### PR TITLE
Use common HTTP server creation logic

### PR DESCRIPTION
**Pull request by ctron**
The removes the gazillion ways we had to create an `HttpServer` instance. All slightly different, all lacking an option here and there. Now all use the same logic.

Remaining tasks:
* [x] In `server`, create a single, combined API endpoint
* [x] Change Helm charts to use new configuration, and create a macro there too
* [x] Allow serving the frontend too

### CODE REVIEW

**Review by lulf — COMMENTED** [review]
Change looks sensible to me. However, I don't see how you intend to use this for simplifying 'server' creating a single http endpoint without a lot of code duplication since it needs to add the API urls, auth config etc, and server would very easily diverge from the microservices.

**Comment by ctron** [comment]
@lulf next version, this time with the composite API, updated helm charts, and an option to serve the UI folder too

**Review by lulf — COMMENTED** [review]
nice!

File: service-common/Cargo.toml [hunk 1]
```diff
@@ -51,6 +54,8 @@ actix-rt = "2.7"
 tokio = { version = "1", features = ["full"] }
 
 [features]
-default = ["rustls", "native-tls"]
+default = ["rustls", "native-tls", "actix", "openssl"]
 rustls = ["rust_tls", "webpki", "reqwest/rustls-tls"]
+openssl = ["open_ssl", "drogue-cloud-service-api/openssl"]
```
[the remark this answers is not in the document]
Comment (ctron) [reply service-common/Cargo.toml:59]: Yea, I wasn't sure how this works exactly. Optional dependencies and features still confuse me.

[the remark this answers is not in the document]
Comment (ctron) [reply service-common/Cargo.toml:59]: Ok, cloud be: https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

I guess with that, I would just override the auto-generated feature, and still add `dep:openssl` in the process.

**Pull request merged**